tuplex/codegen/include/IteratorContextProxy.h

```diff
@@ -0,0 +1,172 @@
+#ifndef TUPLEX_ITERATORCONTEXTPROXY_H
+#define TUPLEX_ITERATORCONTEXTPROXY_H
+
+#include <llvm/IR/IRBuilder.h>
+#include <LambdaFunction.h>
+#include <LLVMEnvironment.h>
+#include <ASTAnnotation.h>
+
+namespace tuplex {
+    namespace codegen{
+        /*!
+         * helper class for iterator-related calls
+         */
+        class IteratorContextProxy {
```

Comment:
Maybe we should create classes for each zip, enumerate etc. inheriting from some base class proxy class?

Reply:
postponed.